VAE.py

```diff
@@ -16,18 +18,11 @@ class Encoder(nn.Module):
     """
 
     def __init__(
-        self,
-        input_dim,
-        latent_dim,
-        hidden_dim=32,
-        activation=nn.Tanh,
-        device="cpu",
+        self, input_dim, latent_dim, hidden_dim=32, activation=nn.Tanh, device="gpu",
```

Comment:
It seems like it would be good to add flexibility to codebase to allow for hidden layers to be passed in a list/tuple e.g. (128, 64) c.f. `sdv` such that we can have multiple hidden layers of different sizes.
This would lead to a setup like 128 -> 64 -> 32 (latent) -> 64 -> 128.
Might be for a follow up PR - maybe add as an issue?